### PR TITLE
Feat: Add spacingX and spacingY to Flex component #DS-1427

### PR DESCRIPTION
**Pull request by curdaj**

### CODE REVIEW

**Review by pavelklibani — APPROVED** [review]
👍 LGTM

File: packages/web-react/src/components/Flex/__tests__/Flex.test.tsx
Comment (literat) [inline packages/web-react/src/components/Flex/__tests__/Flex.test.tsx:1]: **polish (non-blocking):** Can't we just use some kind of data provider like `it.each`?

File: packages/web-react/src/components/Grid/useGridStyleProps.ts [hunk 1]
```diff
@@ -23,10 +23,8 @@ export function useGridStyleProps(props: SpiritGridProps<ElementType>): GridStyl
   const gridClass = useClassNamePrefix('Grid');
 
   const gridStyle: GridCSSProperties = {
-    ...{
-      ...useSpacingStyle(spacing, 'grid', DirectionAxis.X),
-      ...useSpacingStyle(spacing, 'grid', DirectionAxis.Y),
-    },
+    ...useSpacingStyle(spacing, 'grid', DirectionAxis.X),
```
Comment (literat) [inline packages/web-react/src/components/Grid/useGridStyleProps.ts:26]: **praise (non-blocking):** Thanks for this polishing 👍

**Review by literat — COMMENTED** [review]
Will there be any change also in the demo or story so these new props will be shown somewhere?

**Comment by curdaj** [comment]
> Will there be any change also in the demo or story so these new props will be shown somewhere?

Demo added

File: packages/web-react/src/components/Flex/__tests__/Flex.test.tsx
Comment (curdaj) [reply packages/web-react/src/components/Flex/__tests__/Flex.test.tsx:1]: Yes, I can do that, but I don't think it's more clear. What you think?

```
type SpacingValue = `space-${number}`;

const testCases = [
  {
    description: 'should render with custom spacing',
    props: { spacing: 'space-600' as SpacingValue },
    expectedStyles: {
      '--flex-spacing-x': 'var(--spirit-space-600)',
      '--flex-spacing-y': 'var(--spirit-space-600)',
    },
  },
  {
    description: 'should render with custom spacingX and spacingY',
    props: { spacingX: 'space-500' as SpacingValue, spacingY: 'space-1600' as SpacingValue },
    expectedStyles: {
      '--flex-spacing-x': 'var(--spirit-space-500)',
      '--flex-spacing-y': 'var(--spirit-space-1600)',
    },
  },
  {
    description: 'should render with custom spacing for each breakpoint',
    props: {
      spacing: { mobile: 'space-100' as SpacingValue, tablet: 'space-1000' as SpacingValue, desktop: 'space-1200' as SpacingValue },
    },
    expectedStyles: {
      '--flex-spacing-x': 'var(--spirit-space-100)',
      '--flex-spacing-y': 'var(--spirit-space-100)',
      '--flex-spacing-x-tablet': 'var(--spirit-space-1000)',
      '--flex-spacing-y-tablet': 'var(--spirit-space-1000)',
      '--flex-spacing-x-desktop': 'var(--spirit-space-1200)',
      '--flex-spacing-y-desktop': 'var(--spirit-space-1200)',
    },
  },
  {
    description: 'should render with custom spacingX and spacingY for each breakpoint',
    props: {
      spacingX: { mobile: 'space-100' as SpacingValue, tablet: 'space-1000' as SpacingValue, desktop: 'space-1200' as SpacingValue },
      spacingY: { mobile: 'space-200' as SpacingValue, tablet: 'space-1100' as SpacingValue, desktop: 'space-1600' as SpacingValue },
    },
    expectedStyles: {
      '--flex-spacing-x': 'var(--spirit-space-100)',
      '--flex-spacing-y': 'var(--spirit-space-200)',
      '--flex-spacing-x-tablet': 'var(--spirit-space-1000)',
      '--flex-spacing-y-tablet': 'var(--spirit-space-1100)',
      '--flex-spacing-x-desktop': 'var(--spirit-space-1200)',
      '--flex-spacing-y-desktop': 'var(--spirit-space-1600)',
    },
  },
];

it.each(testCases)('$description', ({ props, expectedStyles }) => {
  render(<Flex {...props} data-testid={testId} />);
  const element = screen.getByTestId(testId) as HTMLElement;

  for (const [styleProp, value] of Object.entries(expectedStyles)) {
    expect(element).toHaveStyle({ [styleProp]: value });
  }
});
```

File: packages/web/src/scss/components/Flex/_Flex.scss [hunk 2]
```diff
@@ -11,8 +11,16 @@
 .Flex {
     @include reset.list();
     @include spacing.create(
-        $input-property-base-name: '--flex-spacing',
-        $output-property-name: '--flex-gap',
```
Comment (adamkudrna) [inline packages/web/src/scss/components/Flex/_Flex.scss:15]: ⚠️ Please mind this is a BC for the web package.

**Review by crishpeen — COMMENTED** [review]
This is a BC in `web` package. We should document this: commit prefix and web Migration Guide.

Please add these props to Storybook.

**Review by adamkudrna — APPROVED** [review]

**Comment by curdaj** [comment]
> This is a BC in `web` package. We should document this: commit prefix and web Migration Guide.
> 
> Please add these props to Storybook.

Commit prefix and Migration Guide for web added

File: packages/web-react/src/components/Flex/__tests__/Flex.test.tsx
Comment (literat) [reply packages/web-react/src/components/Flex/__tests__/Flex.test.tsx:1]: Well, it looks way better. Now I can see the test cases without the boilerplate of repeating `expect`. I like how you are using the keys of the object to make it more descriptive 👍

**Review by literat — APPROVED** [review]
Good Job 👍

File: docs/migrations/web/MIGRATION-v3.md [hunk 3]
```diff
@@ -210,6 +211,25 @@ Button `inverted` modifier was removed. Use themes to switch the button colors.
...
+
+Similarly, when applying responsive spacing for a specific breakpoint, the previous property `--flex-spacing-{breakpoint}` has been replaced.
+You should now use `--flex-spacing-x-{breakpoint}` for horizontal spacing or `--flex-spacing-y-{breakpoint}` for vertical spacing at that specific breakpoint.
+
+#### Migration Guide
+
+To separate spacing on the x and y axes, please manually replace the original `--flex-spacing` property with `--flex-spacing-x` or `--flex-spacing-y`,
+depending on the axis you want to apply the spacing to.
+
+- `<div style="--flex-spacing: var(--spirit-space-1200)">` → `<div style="--flex-spacing-x: var(--spirit-space-1200)">`
```
Comment (crishpeen) [inline docs/migrations/web/MIGRATION-v3.md:227]: I would add `class="Flex …"` so they know on which element it should be.

```suggestion
- `<div class="Flex …" style="--flex-spacing: var(--spirit-space-1200)">` → `<div class="Flex …" style="--flex-spacing-x: var(--spirit-space-1200)">`
```
What do you think?

File: docs/migrations/web/MIGRATION-v3.md [hunk 4]
```diff
@@ -210,6 +211,25 @@ Button `inverted` modifier was removed. Use themes to switch the button colors.
...
+
+The original property `--flex-spacing` has been replaced to allow separate control over spacing on the x and y axes.
+You can now use the properties `--flex-spacing-x` and `--flex-spacing-y` to apply spacing only on the x-axis or y-axis, respectively.
+
+Similarly, when applying responsive spacing for a specific breakpoint, the previous property `--flex-spacing-{breakpoint}` has been replaced.
+You should now use `--flex-spacing-x-{breakpoint}` for horizontal spacing or `--flex-spacing-y-{breakpoint}` for vertical spacing at that specific breakpoint.
+
+#### Migration Guide
+
+To separate spacing on the x and y axes, please manually replace the original `--flex-spacing` property with `--flex-spacing-x` or `--flex-spacing-y`,
```
[the remark this answers is not in the document]
Comment (crishpeen) [reply docs/migrations/web/MIGRATION-v3.md:224]: or and/or as you use below

File: packages/web-react/src/components/Flex/stories/Flex.stories.tsx [hunk 5]
```diff
@@ -63,6 +63,16 @@ const meta: Meta<typeof Flex> = {
       tablet: 'space-800',
       desktop: 'space-1000',
     },
+    spacingX: {
```
Comment (crishpeen) [inline packages/web-react/src/components/Flex/stories/Flex.stories.tsx:66]: please add this to `argTypes` too.

```
    spacingX: {
      control: 'object',
    },
    spacingY: {
      control: 'object',
    },
```

**Review by crishpeen — APPROVED** [review]

**Pull request merged**